src/ray/gcs/gcs_server/gcs_health_check_manager.h

```diff
@@ -124,7 +124,7 @@ class GcsHealthCheckManager {
    private:
     void StartHealthCheck();
 
-    GcsHealthCheckManager *manager_;
+    std::shared_ptr<GcsHealthCheckManager> manager_;
```

Comment:
weak_ptr is better here (better indicates the ownership), this shared_ptr + weak_ptr pattern is used in retryable_grpc_client

Reply:
No strong opinion, otherwise you need to check before usage at check. Updated anyway.